src/store/Store.ts

```diff
@@ -25,12 +25,15 @@ export interface Warning {
   message: string;
 }
 
+export type RenderMode = "static" | "dynamic";
```

Comment:
use enum will be better, so that you don't need to hard code as bellow,

```JS
mode === 'dynamic'

store.state.mode === "static"
```

Reply:
Actually using Enum is not a good practice.

1. Assigned value is not stable. It could leave vulnerability to change of code
```ts
// Fist
enum Animal {
	Dog, //0
	Cat, //1
}

// After change of code
enum Animal {
	Bird, // 0
	Dog, //1
	Cat, //2
}
```

2. It does not adhere to DRY principle
```ts
enum Animal {
	Dog = 'Dog',
	Cat = 'Cat',
}
```

3. Lookup and reverse lookup lead to verbosity
```ts
enum Animal {
	Dog = 0,
	Cat = 1,
}

console.log(Animal['Dog']); // 0
console.log(Animal[0]); // Cat

// Transpiled code 
(function (Animal) {
    Animal[Animal["Dog"] = 0] = "Dog";
    Animal[Animal["Cat"] = 1] = "Cat";
})(Animal || (Animal = {}));